src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -304,13 +305,24 @@ class Host(RESTController):
                  },
                  responses={200: LIST_HOST_SCHEMA})
     @RESTController.MethodMap(version=APIVersion(1, 2))
-    def list(self, sources=None, facts=False):
+    def list(self, sources=None, facts=False, offset: int = 0, 
+             limit: int = 5, search: str = '', sort: str = ''):
         hosts = get_hosts(sources)
+        params = ['hostname']
+        paginator = ListPaginator(int(offset), int(limit), sort, search, hosts,
+                                  searchable_params=params, sortable_params=params,
+                                  default_sort='+hostname')
+        hosts = [host for host in paginator.list()]
```

Comment:
Don't reuse variable names... That makes code less readable and confusing. What about `paginated_hosts` or `hosts_rage` or... maybe we don't even need the first `hosts`.